### PR TITLE
Feat : Allow to be notified of unconfirmed/confirmed fetched transaction

### PR DESCRIPTION
**Pull request by Alex-Werner**
### Issue being fixed or implemented
- Ability to use Wallet-lib to be notified on new fetched transactions on the network. This intended to be used for better integration when incoming transactions.

### What was done
- Added `FETCHED_UNCONFIRMED_TRANSACTION`, `FETCHED_CONFIRMED_TRANSACTION` events
- Impr : Internal fetched address to also have index
- Impr : Propagate network to Storage
- Doc : Events were missing from the doc, fixed !

### Notes
This goes on top of #45

### CODE REVIEW

File: docs/account.md [hunk 1]
```diff
@@ -171,3 +173,37 @@ const txHistory = account.getTransactionHistory();
 ```
 account.disconnect();
 ```
+
+### Events 
+
+```json
```
Comment (stpdenis) [inline docs/account.md]: javascript

File: docs/account.md [hunk 2]
```diff
@@ -171,3 +173,37 @@ const txHistory = account.getTransactionHistory();
...
+const {EVENTS} = require('@dashevo/wallet-lib');
+const {READY} = EVENTS;
+const doSomethingWhenReady = (info) => {...}
+account.events.on(READY, doSomethingWhenReady);
+
+```
+
+Events types :
```
Comment (stpdenis) [inline docs/account.md]: Would be cool to put it in a table
https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#tables

File: docs/wallet.md [hunk 3]
```diff
@@ -90,4 +92,4 @@ const mnemonic = wallet.exportWallet();
 
 ```
```
Comment (stpdenis) [inline docs/wallet.md]: javascript

File: src/Storage/importTransaction.js [hunk 4]
```diff
@@ -1,23 +1,26 @@
...
  */
-const importTransaction = function (transaction) {
+const importTransaction = function importTransaction(transaction) {
   const self = this;
   if (!is.transactionObj(transaction)) throw new InvalidTransactionObject(transaction);
-  const transactionStore = this.store.transactions;
+  const { store } = this;
+
+  const transactionStore = store.transactions;
+  const currBlockheight = store.chains[this.network].blockheight;
```
Comment (stpdenis) [inline src/Storage/importTransaction.js]: I think if we use this/self, it should be decomposed in the beginning of the function, so dependencies are more transparent.

File: src/Storage/importTransaction.js [hunk 5]
```diff
@@ -54,8 +57,21 @@ const importTransaction = function (transaction) {
       }
     });
     this.lastModified = +new Date();
+
+
+    const secureBlockheight = (transaction.blockheight + SECURE_TRANSACTION_CONFIRMATIONS_NB);
```
Comment (stpdenis) [inline src/Storage/importTransaction.js]: Why braces here?

File: src/plugins/Workers/SyncWorker.js [hunk 6]
```diff
@@ -104,22 +104,35 @@ class SyncWorker extends Worker {
...
         walletPaths.forEach((path) => {
           const address = walletAddresses[path];
-          if (address.unconfirmedBalanceSat > 0
-            || address.fetchedLast < Date.now() - self.fetchThreshold) {
+
+
+          const hasUnconfirmedBalance = (address.unconfirmedBalanceSat > 0);
+          // This will make all last address (the one we got from unconfirmedAddress)
+          // to basically check each time
+          const isFirstAndUnused = (address.used === false && address.index === 0);
```
Comment (stpdenis) [inline src/plugins/Workers/SyncWorker.js]: why braces?

File: src/plugins/Workers/SyncWorker.js [hunk 7]
```diff
@@ -104,22 +104,35 @@ class SyncWorker extends Worker {
...
+          const hasMostChanceToReceiveTx = ((prevWasUsed === true && address.used === false));
```
Comment (stpdenis) [inline src/plugins/Workers/SyncWorker.js]: double braces?

File: docs/account.md (hunk 2 above)
Comment (Alex-Werner) [reply docs/account.md]: Done :)

File: docs/account.md (hunk 1 above)
Comment (Alex-Werner) [reply docs/account.md]: Done

File: src/Storage/importTransaction.js (hunk 5 above)
Comment (Alex-Werner) [reply src/Storage/importTransaction.js]: Resolved

File: src/plugins/Workers/SyncWorker.js (hunk 6 above)
Comment (Alex-Werner) [reply src/plugins/Workers/SyncWorker.js]: Resolved

File: src/plugins/Workers/SyncWorker.js (hunk 7 above)
Comment (Alex-Werner) [reply src/plugins/Workers/SyncWorker.js]: Resolved

File: src/Storage/importTransaction.js (hunk 4 above)
Comment (Alex-Werner) [reply src/Storage/importTransaction.js]: Resolved.

**Pull request merged**